mentorship ios/ViewModels/RelationViewModel.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  RelationViewModel.swift
+//  Created on 02/07/20
+//  Created for AnitaB.org Mentorship-iOS 
+//
+
+import Foundation
+import Combine
+
+class RelationViewModel: ObservableObject {
+    
+    // MARK: - Variables
+    @Published var currentRelation = RelationModel().currentRelation
+    @Published var responseData = RelationModel.ResponseData(message: "")
+    var tasks = RelationModel().tasks
+    @Published var newTask = RelationModel.AddTaskData(description: "")
+    @Published var toDoTasks = RelationModel().tasks
+    @Published var doneTasks = RelationModel().tasks
+    @Published var inActivity = false
+    @Published var addTask = false
+    @Published var personName = ""
+    static var taskTapped = RelationModel().task
+    private var cancellable: AnyCancellable?
+    private var tasksCancellable: AnyCancellable?
+    
+    // MARK: - Functions
+    init() {
+        fetchCurrentRelation()
+    }
+    
+    func fetchCurrentRelation() {
+        //get auth token
+        guard let token = try? KeychainManager.getToken() else {
+            return
+        }
+        
+        //set inActivity
+        inActivity = true
+        
+        //api call
+        cancellable = NetworkManager.callAPI(urlString: URLStringConstants.MentorshipRelation.currentRelation, token: token)
+            .receive(on: RunLoop.main)
+            .catch { _ in Just(self.currentRelation) }
+            .sink { [weak self] current in
+                //use current relation data
+                self?.currentRelation = current
+                self?.personName = self?.getPersonName(data: current) ?? ""
+                self?.inActivity = false
+                //chain api call. get current tasks using id from current relation
+                self?.fetchTasks(id: current.id ?? 0, token: token)
+        }
+    }
+    
+    func fetchTasks(id: Int, token: String) {
+        self.tasksCancellable = NetworkManager.callAPI(urlString: URLStringConstants.MentorshipRelation.getCurrentTasks(id: id), token: token)
+            .receive(on: RunLoop.main)
+            .catch { _ in Just(self.tasks) }
+            .sink { [weak self] tasks in
+                if NetworkManager.responseCode == 200 {
```

Comment:
How are you handling error if the returned response code is not 200. 
I have tested it, it looks good. But in scenarios like, when sever is down.